app/models/pusher.rb

```diff
@@ -44,12 +42,15 @@ def save
   end
 
   def pull_spec
-    GemPackage.open body, "r", nil do |pkg|
-      @spec = pkg.metadata
-      return true
+    gem_tar = Gem::Package::TarReader.new body
```

Comment:
This should switch to `Gem::Package.new(body).load_spec` when rubygems.org is upgraded to RubyGems 2.3 or newer, can we leave a comment?